server/routes/incidents_routes.py

```
@@ -2027,6 +2027,47 @@ def get_recent_unlinked_incidents(user_id):
+@incidents_bp.route("/api/incidents/<incident_id>/action-runs", methods=["GET"])
+@require_permission("incidents", "read")
+def get_incident_action_runs(user_id, incident_id: str):
+    """Return action runs linked to a specific incident."""
+    if not _validate_uuid(incident_id):
+        return jsonify({"error": "Invalid incident ID"}), 400
+
+    org_id = get_org_id_from_request()
+
+    try:
+        with db_pool.get_admin_connection() as conn:
+            with conn.cursor() as cursor:
```

Ruff (0.15.15)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

Combine `with` statements

(SIM117)